Loop/Managers/DeviceAlert/DeviceAlertManager.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  DeviceAlertManager.swift
+//  Loop
+//
+//  Created by Rick Pasetto on 4/9/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import LoopKit
+
+/// Main (singleton-ish) class that is responsible for:
+/// - managing the different targets (handlers) that will post alerts
+/// - serializing alerts to storage
+/// - etc.
+public final class DeviceAlertManager {
+
+    let handlers: [DeviceAlertHandler]
+    var responders: [String: Weak<DeviceAlertResponder>] = [:]
+    
+    public init(rootViewController: UIViewController,
+                isAppInBackgroundFunc: @escaping () -> Bool) {
+        handlers = [UserNotificationDeviceAlertHandler(isAppInBackgroundFunc: isAppInBackgroundFunc),
+                    InAppModalDeviceAlertHandler(rootViewController: rootViewController)]
+    }
+    
+    public func addAlertResponder(key: String, alertResponder: DeviceAlertResponder) {
+        responders[key] = Weak(alertResponder)
+    }
+    
+    public func removeAlertResponder(key: String) {
+        responders.removeValue(forKey: key)
+    }
+    
+    public func acknowledgeDeviceAlert(managerIdentifier: String, alertTypeIdentifier: DeviceAlert.TypeIdentifier) {
+        if let responder = responders[managerIdentifier]?.value {
+            responder.acknowledgeAlert(typeIdentifier: alertTypeIdentifier)
```

Comment:
Note: it isn't logging the device event here (anymore), but it is probably redundant anyway.